application/backend/src/main/resources/apps/typerefinery/components/structure/page/clientlibs/functions.js

```diff
@@ -3,9 +3,12 @@ window.Typerefinery.Components = Typerefinery.Components || {};
 window.Typerefinery.Theme = Typerefinery.Theme || {};
 window.MessageService = window.MessageService || {};
 window.MessageService.Client = MessageService.Client || {};
+window.Typerefinery.Components.Widgets = Typerefinery.Components.Widgets || {};
```

Comment:
this namespace does not match what the file is. This namespace should be Typerefinery.Components.Page meaning all functions associated with Page, so components could call these functions to get page level functionality.